crux-core/src/crux/bootstrap.clj

```diff
@@ -115,13 +115,24 @@
     (tx/await-tx-time indexer tx-time (when timeout {:crux.tx-log/await-tx-timeout (.toMillis timeout)})))
 
   backup/ISystemBackup
-  (write-checkpoint [this {:keys [crux.backup/checkpoint-directory]}]
-    (kv/backup kv-store (io/file checkpoint-directory "kv-store")))
+  (write-checkpoint [this {:keys [crux.backup/checkpoint-directory] :as opts}]
+    (kv/backup kv-store (io/file checkpoint-directory "kv-store"))
+    (when (satisfies? tx-log backup/ISystemBackup)
+      (backup/write-checkpoint tx-log opts)))
 
   Closeable
   (close [_]
-    (some-> close-promise (deliver true))
-    (some-> node-thread (.join))))
+    (when close-fn (close-fn))))
+
+(defn start-crux-node [& bindings]
```

Comment:
Would prefer this to take a map and not do the last/partition stuff. Otherwise this PR looks good.

Reply:
Making a commit to address this.